tests/conftest.py

```diff
@@ -1,43 +1,17 @@
-import os
 
-import pytest
-from sqlalchemy.exc import ProgrammingError
-
-from ..log import LogConfiguration
-from ..model import (SessionManager,
-    Base,
-)
-
-
-@pytest.fixture(autouse=True, scope="session")
-def session_fixture():
-    # This will make sure we always connect to the test database.
-    os.environ['TESTING'] = 'true'
-
-    # Ensure that the log configuration starts in a known state.
-    LogConfiguration.initialize(None, testing=True)
-
-    # Drop any existing schema. It will be recreated when
-    # SessionManager.initialize() runs.
-    #
-    # Base.metadata.drop_all(connection) doesn't work here, so we
-    # approximate by dropping every item individually.
-    engine = SessionManager.engine()
-    for table in reversed(Base.metadata.sorted_tables):
-        statement = table.delete()
-        try:
-            engine.execute(statement)
-        except ProgrammingError as e:
-            # TODO PYTHON3
-            # if isinstance(e.orig, UndefinedTable):
-            if 'does not exist' in e.message:
-                # This is the first time running these tests
-                # on this server, and the tables don't exist yet.
-                pass
-            else:
-                raise
-
-    yield
-
-    if 'TESTING' in os.environ:
-        del os.environ['TESTING']
+# This is kind of janky, but we import the session fixture
+# into these tests here. Plugins need absolute import paths
+# and we don't have a package structure that gives us a reliable
+# import path, so we construct one.
+# todo: reorg core file structure so we have a reliable package name
+from os.path import abspath, dirname, basename
+
+# Pull in the session_fixture defined in core/testing.py
+# which does the database setup and initialization
+pytest_plugins = ["{}.testing".format(basename(dirname(dirname(abspath(__file__)))))]
```

Comment:
I don't really like this, but it does do the job. If anyone has any suggestions for something cleaner here. I can give them a try.

Pytest can only pull in plugins from an absolute import. Because the folder name can change for core, depending what its checked out as, or if its being pulled into circulation etc. 

This lets you run the tests on core from within circulation, or when checked out individually, but figuring out what the package name should be.

Reply:
I don't have a better idea--the issue you describe has definitely caused us problems in the past.